src/main/fc/config.c

```diff
@@ -271,8 +271,9 @@ static void validateAndFixConfig(void)
 #endif // USE_SOFTSPI
 
 #if defined(USE_ADC)
-    if (featureIsEnabled(FEATURE_RSSI_ADC)) {
-        rxConfigMutable()->rssi_channel = 0;
+    if (featureIsEnabled(FEATURE_RSSI_ADC) && rxConfig()->rssi_channel) {
+        featureDisable(FEATURE_RSSI_ADC);
+        rxUpdateRssiSource();
```

Comment:
What is the point of this? `validateAndFixConfig()` is called when the config is first loaded, before the RX is initialised, so the code in `rxUpdateRssiSource()` will be executed twice now.

Reply:
The osd won't update unless this call is made.  The alternative is a manual power cycle.

Reply:
If this is too much trouble, I can just close it.

Reply:
The OSD is currently problematic in the respect that it does not enforce reboots even for changes that need them - we'll probably have to add a facility to enforce reboots from the OSD, and then go through all of the settings and determine if they can safely be done without a reboot.
If this can be changed from the OSD, and should be changeable without a reboot, then the call to `rxUpdateRssiSource()` will have to be outside of `validateAndFixConfig()`, otherwise it will not work as expected in cases where the configuration is changed in OSD but does **not** trigger a change in `validateAndFixConfig()`.

As for closing this, I am not sure - my gut feeling is that this will trip up some users no matter which way we do it - if they select both ADC RSSI and RSSI over an RC channel, one of the two will be disabled on save, and users expecting it to work on this source will report it as broken.
We have had it so that ADC was 'dominant' for a while now (based on a more or less arbitrary decision), and we have not had many users reporting issues due to misconfiguration, so I am not sure if changing this will make it better, or will just get users to report issues because setting up ADC RSSI (and accidentally enabling RSSI over an RX channel as well) stopped working for them.